geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/RasterRDD.scala

```diff
@@ -70,16 +72,27 @@ abstract class RasterRDD[K: AvroRecordCodec: ClassTag] extends TileRDD[K] {
   /** Encode RDD as Avro bytes and return it with avro schema used */
   def toAvroRDD(): (JavaRDD[Array[Byte]], String) = PythonTranslator.toPython(rdd)
 
-  // TODO: get rid of this overload, just use the String interface
-  protected def collectMetadata(crs: Option[CRS], extent: Option[Extent], layout: Option[TileLayout], tileSize: Int): String
+  def collectMetadata(
+    extent: java.util.Map[String, Double],
+    layout: java.util.Map[String, Int],
+    crs: String
+  ): String = {
+    val layoutDefinition = Right(LayoutDefinition(extent.toExtent, layout.toTileLayout))
+
+    collectMetadata(layoutDefinition, crs)
+  }
 
-  def collect_metadata(crs: String, extent: String, layout: String, tileSize: Int): String =
-    collectMetadata(
-      crs = Option(crs).flatMap(TileRDD.getCRS),
-      extent = Option(extent).flatMap(_.parseJson.convertTo[Option[Extent]]),
-      layout = Option(layout).flatMap(_.parseJson.convertTo[Option[TileLayout]]),
-      tileSize = tileSize
-    )
+  def collectMetadata(tileSize: String, crs: String): String = {
+    val layoutScheme =
+      if (tileSize != "")
+        Left(FloatingLayoutScheme(tileSize.toInt))
+      else
+        Left(FloatingLayoutScheme())
+
+    collectMetadata(layoutScheme, crs)
+  }
+
+  def collectMetadata(layout: Either[LayoutScheme, LayoutDefinition], crs: String): String
```

Comment:
crs: should be `CRS` since we're making separation between scala and python APIs